Sources/Apodini/Request/Request.swift

```diff
@@ -5,28 +5,64 @@
 //  Created by Paul Schmiedmayer on 7/12/20.
 //
 
-import Vapor
+import Foundation
+import NIO
+@_implementationOnly import Vapor
 
+/// A type representing a server response.
+public protocol Request {
+    /// parameters
+    func parameter<T: Codable>(for parameter: Parameter<T>) throws -> T
+    /// The EventLoop associated with this request
+    var eventLoop: EventLoop { get }
+    /// database
+    var database: Any? { get }
+}
+
+typealias VaporRequest = Vapor.Request
+
+extension VaporRequest: Request {
+    public func parameter<T: Codable>(for parameter: Parameter<T>) throws -> T {
+
+        switch parameter.option(for: PropertyOptionKey.http) {
+        case .path where T.self is LosslessStringConvertible:
+            guard let parameter = parameters.get(parameter.id.uuidString) as? T else {
+                throw Vapor.Abort(.badRequest)
+            }
+            return parameter
+        case .query:
+            guard let name = parameter.name else {
+                throw Vapor.Abort(.badRequest)
+            }
+            return try query.get(T.self, at: name)
+        case .body:
+            return try content.decode(T.self)
+        default:
+            throw Vapor.Abort(.badRequest)
```

Comment:
The idea is that when building the semantic model we decide how to decode it (and infer the best strategy, for now we can default to something else), and store that information paired to the ID of the Parameter.

Reply:
A first implementation of `@Parameter` and its inference strategies are already part of the shared semantic model #31. It may be worth coordinating how both of them get merged 🤔